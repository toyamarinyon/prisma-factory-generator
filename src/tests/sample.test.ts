import { getDMMF } from '@prisma/sdk'
import {
  addModelFactoryDeclaration,
  addPrismaImportDeclaration,
} from '../generator'
import { Project, ts } from 'ts-morph'
const datamodel = /* Prisma */ `

datasource db {
  ///sqlite doesn't support String[]
  provider = "postgresql"
  url="postgresql://test:test@localhost:5432/test"
}

model User {
  id                Int          @id @default(autoincrement())
  email             String       @unique
  stringArray       String[]       
  jsonProp          Json
  status            UserStatus
  accessToken       AccessToken?

  @@map(name: "users")
}

enum UserStatus {
  active
}

model AccessToken {
  id          Int       @id @default(autoincrement())
  user        User      @relation(fields: [userId], references: [id])
  userId      Int       @unique @map(name: "user_id")

  @@map("access_tokens")
}
`

// test('dmmf snapshot', async() => {
//   const dmmf = await getDMMF({ datamodel })
//   expect(dmmf).toMatchSnapshot()
// })
test('generate model', async () => {
  const project = new Project({})
  const sourceFile = project.createSourceFile('./src/test-src.ts')
  const dmmf = await getDMMF({ datamodel })
  addPrismaImportDeclaration(sourceFile)
  dmmf.datamodel.models.forEach((model) => {
    addModelFactoryDeclaration(
      sourceFile,
      model,
      dmmf.datamodel.models,
      dmmf.datamodel.enums
    )
  })

  sourceFile.formatText({
    indentSize: 2,
    semicolons: ts.SemicolonPreference.Remove,
  })
  expect(sourceFile.getFullText()).toMatchSnapshot()
})
