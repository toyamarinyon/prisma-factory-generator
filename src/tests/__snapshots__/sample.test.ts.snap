// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate model 1`] = `
"import { PrismaClient } from \\"@prisma/client\\"
import { Prisma } from \\".prisma/client\\"
import * as faker from \\"faker\\"
const prisma = new PrismaClient()

export function inputsForUser() {
  return {
    email: faker.internet.email(),
  }
}

type UserFactoryArgs = Omit<Prisma.UserCreateArgs, 'data'> & {
  data: Pick<Prisma.UserCreateInput, 'accessToken'> & Partial<Omit<Prisma.UserCreateInput, 'accessToken'>>
}

export async function createUser(args?: UserFactoryArgs) {

  return await prisma.user.create({
    ...args,
    data: {
      ...inputsForUser(),
      ...args?.data
    },
  })

}

export function inputsForAccessToken() {
  return {
  }
}

type AccessTokenFactoryArgs = Omit<Prisma.AccessTokenCreateArgs, 'data'> & {
  data: Pick<Prisma.AccessTokenCreateInput, 'user'> & Partial<Omit<Prisma.AccessTokenCreateInput, 'user'>>
}

export async function createAccessToken(args: AccessTokenFactoryArgs) {

  return await prisma.accessToken.create({
    ...args,
    data: {
      ...inputsForAccessToken(),
      ...args?.data
    },
  })

}
"
`;
